kat/src/ecc384_kat.rs

```diff
@@ -51,7 +51,7 @@ impl Ecc384Kat {
     /// This function executes the Known Answer Tests (aka KAT) for ECC384.
     ///
     /// Test vector source:
-    /// Generated using MbedTLS library.
+    /// TODO: Verify these vectors against some third party tool
```

Comment:
I have verified they are correct using the Python `cryptography` library (which uses OpenSSL):

```
from cryptography.hazmat.primitives.asymmetric import ec
from cryptography.hazmat.primitives.hashes import SHA384
from cryptography.hazmat.primitives.asymmetric.utils import decode_dss_signature
from cryptography.hazmat.primitives.asymmetric.utils import Prehashed


def to_num(l):
    return int("".join(["%08x" % x for x in l]), 16)

priv_key = to_num([0xfeeef554, 0x4a765649, 0x90128ad1, 0x89e873f2, 0x1f0dfd5a, 0xd7e2fa86, 0x1127ee6e, 0x394ca784, 0x871c1aec, 0x032c7a8b, 0x10b93e0e, 0xab8946d6])
public_x = to_num([0xd7dd94e0, 0xbffc4cad, 0xe9902b7f, 0xdb154260, 0xd5ec5dfd, 0x57950e83, 0x59015a30, 0x2c8bf7bb, 0xa7e5f6df, 0xfc168516, 0x2bdd35f9, 0xf5c1b0ff])
public_y = to_num([0xbb9c3a2f, 0x061e8d70, 0x14278dd5, 0x1e66a918, 0xa6b6f9f1, 0xc1937312, 0xd4e7a921, 0xb18ef0f4, 0x1fdd401d, 0x9e771850, 0x9f8731e9, 0xeec9c31d])
r = to_num([0x93799D55, 0x12263628, 0x34F60F7B, 0x945290B7, 0xCCE6E996, 0x01FB7EBD, 0x026C2E3C, 0x445D3CD9, 0xB65068DA, 0xC0A848BE, 0x9F0560AA, 0x758FDA27])
s = to_num([0xE548E535, 0xA1CC600E, 0x133B5591, 0xAEBAAD78, 0x054006D7, 0x52D0E1DF, 0x94FBFA95, 0xD78F0B3F, 0x8E81B911, 0x9C2BE008, 0xBF6D6F4E, 0x4185F87D])

priv = ec.derive_private_key(priv_key, ec.SECP384R1())
pub = priv.public_key()


assert pub.public_numbers().x == public_x
assert pub.public_numbers().y == public_y

sign_data = b"\x00" * 48
signature = decode_dss_signature(
    priv.sign(sign_data, ec.ECDSA(Prehashed(SHA384()), deterministic_signing=True))
)

assert r == signature[0]
assert s == signature[1]
```

Reply:
I really appreciate you doing this! This python wrapper looks like a great way to leverage OpenSSL for this kind of stuff. I'll update this todo before it merges.